### DISCUSSION TITLE
Understanding "but not" operator on a relation

### DISCUSSION THREAD

**Discussion by Raid55** [openfga/community#240] [Q&A] [answered]
In the DSL Model below:
```
type user

type workspace
  relation member: [user]
  relation guest: [user]

type document
  relations
    define workspace: [workspace]
    define owner: [user] but not guest from workspace # <-- this line specifically
    define full_access: [user, workspace#member] or owner
    define edit: [user, workspace#member] or full_access
    define view: [user, user:*, workspace#member] or edit
```
Does the `owner` relation on the `document` type ensure that a `guest` of the parent workspace can't be written (using `.write`) as an `owner` of a `document` or does this just ensure that in case that tuple exists a `.check` call will deny access?

**Comment by rhamzeh** [comment, marked as answer]
Hi @Raid55 

It affects the `Check` - so if tuples are written to indicate someone is both an owner and a guest, the check for owner will return false.

If you want to have a check on write, you should treat it as any other action:
- add a relation called `can_be_written_as_owner`
- before calling the `Write`, call `Check` to ensure that the action of writing can proceed
